docs/seed-image-generation.md

```diff
@@ -0,0 +1,329 @@
+# IBU Seed Image Generation
+
+- [IBU Seed Image Generation](#ibu-seed-image-generation)
+  - [Overview](#overview)
+  - [Seed SNO Pre-Requisites](#seed-sno-pre-requisites)
+    - [Shared Container Storage](#shared-container-storage)
+    - [Required dnsmasq Configuration](#required-dnsmasq-configuration)
+  - [SeedGenerator CR](#seedgenerator-cr)
+    - [Creating the seedgen Secret CR](#creating-the-seedgen-secret-cr)
+    - [Creating the seedimage SeedGenerator CR](#creating-the-seedimage-seedgenerator-cr)
+  - [Generating the IBU Seed Image](#generating-the-ibu-seed-image)
+    - [Monitoring Progress](#monitoring-progress)
+  - [ACM and ZTP GitOps Considerations](#acm-and-ztp-gitops-considerations)
+
+## Overview
+
+The Lifecycle Agent provides orchestration of IBU Seed Image generation via the `SeedGenerator` CRD. As part of orchestration, the LCA will:
+
+- Perform system configuration checks to ensure any required configuration is present.
+- Perform any necessary system cleanup prior to generating the seed image.
+- Launch the ibu-imager tool, which will:
+  - Shutdown the cluster operators
+  - Prepare seed image config
+  - Generate and publish the seed image
+  - Restore the cluster operators
+- (not yet implemented) After LCA operator recovers, restore the seedgen CR and update its status to reflect success or failure of image generation, and restore the `ManagedCluster` CR on the hub (if applicable).
+
+## Seed SNO Pre-Requisites
+
+The seed SNO configuration has some pre-requisites:
+
+- The CPU topology must align with the target SNO(s).
+  - Same number of cores.
+  - Same tuned performance configuration (ie. reserved CPUs).
+  - FIPS enablement (? *TODO*: confirm)
+- Deployed in the same manner as target SNO(s).
+  - Same ACM/MCE version.
+    - Exception: Hub for seed SNO must not have extra ACM addons enabled (ie. observability). The LCA orchestration confirms that no such addons are present on the seed SNO as part of its system config validation.
```

Comment:
Should we mention some info on when seed SNO is deployed with ZTP? i.e when deployed with ZTP we should just  comment our siteconfig from kustomize to get the best result